pkg/cloudprovider/fake/cloudprovider.go

```diff
@@ -32,43 +32,45 @@ import (
 
 type CloudProvider struct{}
 
-func (c *CloudProvider) Create(ctx context.Context, constraints *v1alpha4.Constraints, instanceTypes []cloudprovider.InstanceType, bind func(*v1.Node) error) chan error {
-	name := strings.ToLower(randomdata.SillyName())
-	// Pick first instance type option
-	instance := instanceTypes[0]
-	// Pick first zone
-	zones := instance.Zones()
-	if len(constraints.Zones) != 0 {
-		zones = functional.IntersectStringSlice(constraints.Zones, instance.Zones())
-	}
-	zone := zones[0]
-
+func (c *CloudProvider) Create(ctx context.Context, constraints *v1alpha4.Constraints, instanceTypes []cloudprovider.InstanceType, quantity int, bind func(*v1.Node) error) chan error {
 	err := make(chan error)
-	go func() {
-		err <- bind(&v1.Node{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: name,
-				Labels: map[string]string{
-					v1.LabelTopologyZone:       zone,
-					v1.LabelInstanceTypeStable: instance.Name(),
+	for i := 0; i < quantity; i++ {
+		name := strings.ToLower(randomdata.SillyName())
+		// Pick first instance type option
```

Comment:
I can remove the comments where it says pick first {instance, zone}. I'm not entirely sure the reason on picking 1 since this was existing code that I just wrapped with the for-loop.  I don't think doing anything more complicated would make a difference to tests that depend on this code. If the real provider were used, it would be semi-random (not really, but allocation strategy would pick one on the Fleet backend).